docs/variants/qemu_q35/building-manual.md

```diff
@@ -0,0 +1,66 @@
+# Dasharo (UEFI) v0.1.0 for QEMU Q35 - Building Manual
+
+Follow below steps to create the "Dasharo (UEFI) OVMF image for QEMU Q35" from EDK2:
+
+## Requirements
+
+- Docker
+    + follow [Install Docker Engine on Ubuntu](https://docs.docker.com/engine/install/ubuntu/)
+    + follow [Post-installation steps for Linux](https://docs.docker.com/engine/install/linux-postinstall/)
+
+## Procedure
+
+Easiest way to build OVMF for QEMU Q35 is to use coreboot Docker image.
+As some dependencies maybe missing in given local instance.
+
+Obtain the docker image:
+
+```bash
+docker pull coreboot/coreboot-sdk:2021-09-23_b0d87f753c
+```
+
+Clone official Dasharo EDK2 repository to your docker instance,
+with git or downloading the source code from github.
+
+```bash
+git clone https://github.com/Dasharo/edk2.git
+```
+
+1. Follow below instructions, to prepare your environment for building OVMF image.
+
+- Setup the environment variables with the following command -
+
+```bash
+make -C BaseTools source edksetup.sh
+```
+
+- Update the submodules in order get latest dependencies.
+And avoid any build process errors while building the OVMF image.
+
+```bash
+git submodules update --init
+```
+
+## Building the firmware image
+
+1. To build firmware image with required features set the flags with **TRUE/FLASE**.
+Following is an example of SMM feature enabled,
+`DEFINE SMM_REQUIRE = TRUE` in the **OvmfPkgX64.dsc** file.
+
+ Check the following build command:
+
+```bash
+build -D DEBUG_ON_SERIAL_PORT -D BOOTLOADER=COREBOOT -a IA32 -a X64 -t GCC5 -b DEBUG -p OvmfPkg/OvmfPkgX64.dsc.
+```
+
+- `DEBUG_ON_SERIAL_PORT` = The following flag must be given for the **DEBUG** build
+as the output is redirected to the I/O port **0x402** to qemu monitor.
```

Comment:
It doesn;t have to be given. Moreover it shouldn't be given, because t breaks the debug output to the file. If we do not want any debug output we should build release image.

Reply:
I thought it's helpful for people who want to debug or want to know how to debug just in case. I will make sure to remove it.

Reply:
I meant:

`DEBUG_ON_SERIAL_PORT` can't be passed to the build command because it breaks the debug output to 0x402 port and the file itself.

So debug output to port 0x402 to a file works only if  `DEBUG_ON_SERIAL_PORT` is not passed. At least this is what I have observed

Reply:
Yes without the DEBUG_ON_SERIAL_PORT command the debug log is generated and can view the logs.